2-Draft-Accepted/RFCxxxx-PowerShellGet-3.0.md

```diff
@@ -0,0 +1,476 @@
+as a differently named module already on the system.
+
+A `-Quiet` switch will suppress progress information.
+
+`-Scope` with `AllUsers` and `CurrentUser` will work the same as it does in v2.
+Default is `CurrentUser`.
+
+If aliases are enabled `Install-Module` will call to `Install-PSResource -Type Module`.
+
+v3 will continue to use the versioning folder format as v2 in that it will be
+Major.Minor.Patch and not contain the semver prerelease label.
+
+### Dependencies and version management
+
+`Install-PSResource` can accept a path to a psd1 or json file (using `-RequiredResourceFile`),
+or a hashtable or json (using `-RequiredResource`) where the key is the module name and the
+value is either the required version specified using Nuget version range syntax or
+a hash table where `repository` is set to the URL of the repository and
+`version` contains the [Nuget version range syntax](https://docs.microsoft.com/en-us/nuget/reference/package-versioning#version-ranges-and-wildcards).
+
+```powershell
+Install-PSResource -RequiredResource @{
+  'Configuration' = '[1.3.1,2.0)'
+  'Pester'        = @{
+    version = '[4.4.2,4.7.0]'
+    repository = 'https://www.powershellgallery.com'
+    credential = $cred
+    allowPrerelease = $true
+  }
+}
+```
+In this case the modules named "Configuration", and "Pester" will be installed. 
+The json format will be the same as if this hashtable is passed to `ConvertTo-Json`:
+
+```json
+{
+  'Configuration': '[1.3.1,2.0)',
+  'Pester': {
+    'version': '[4.4.2,4.7.0]',
+    'credential': null,
+    'repository': 'https://www.powershellgallery.com',
+    'prerelease': true
+  }
+}
+```
+
+>[!NOTE]
+>Credentials management will be using [Secrets Management RFC](https://github.com/PowerShell/PowerShell-RFC/pull/208)
+> for generalized credential management.
+The `repository` property can be the name of the registered repository or the URL
+to a repository.
+If no `repository` is specified, it will use the `default` repository.
+
+The older `System.Version` four part version type will be supported to retain
+compatibility with existing published modules using that format.
+
+If the resource requires dependencies that are not already installed, then
+a prompt will appear _before anything is installed_, listing all the resources
+including module name, version, size, and repository to be installed unless
+`-IncludeDependencies` is specified which will install without prompting.
+Rejecting the prompt will result in nothing being installed.
+
+Declared dependencies are searched and installed using the same trust algorithm
+as described for `Install-PSResource` above.
+
+>[!NOTE]
+>Installing dependencies is following the `apt` experience in that it prompts
+>by default instead of automatically installing dependencies.  Since dependencies
+>can be quite large, this would be equivalent to `ConfirmImpact=High` which
+>would prompt by default.
+>Dependency installation works with `-DestinationPath` parameter.

```

Comment:
This needs to have the ability to push though for CI via use of either the force parameter or -Confirm:$false but should not need both